R/classicalpredictionperformance.R

```diff
@@ -0,0 +1,654 @@
+  funnelTestTable$addColumnInfo(name = "df",     title = gettext("df"),           type = "integer")
+  funnelTestTable$addColumnInfo(name = "p",      title = "p",                     type = "pvalue")
+  jaspResults[["funnelTestTable"]] <- funnelTestTable
+
+  if (is.null(fatFits) || jaspResults[["summaryTable"]]$getError())
+    return()
+
+  for(i in seq_along(fatFits)) {
+    if (inherits(fatFits[[i]], c("simpleError", "error"))) {
+      funnelTestTable$addRows(list(
+        method  = .metamiscFitFunnelAsymmetryNames(names(fatFits)[i])
+      ))
+      funnelTestTable$addFootnote(gettextf("The %1$s test failed with the following error: %2$s",
+                                           .metamiscFitFunnelAsymmetryNames(names(fatFits)[i]),
+                                           fatFits[[i]]$message))
+    } else {
+      funnelTestTable$addRows(list(
+        method  = .metamiscFitFunnelAsymmetryNames(fatFits[[i]]$method),
+        t       = fatFits[[i]]$tval,
+        df      = fatFits[[i]]$df,
+        p       = fatFits[[i]]$pval
+      ))
+    }
+
+  }
+
+
+  return()
+}
+.metamiscFitFunnelAsymmetryPlot  <- function(jaspResults, options) {
+
+  if (!is.null(jaspResults[["funnelTestPlots"]])) {
+    funnelTestPlots <- jaspResults[["funnelTestPlots"]]
+  } else{
+    funnelTestPlots <- createJaspContainer(title = gettext("Funnel Plot Asymmetry Plots"))
+    funnelTestPlots$dependOn(c(.metamiscDependencies, "funnelAsymmetryTestPlot", "funnelAsymmetryTest", if (options[["method"]] == "BAYES") .metamiscDependenciesBayesian))
+    funnelTestPlots$position <- 4
+    jaspResults[["funnelTestPlots"]] <- funnelTestPlots
+  }
+
+  fatFits <- jaspResults[["modelsFat"]]$object
+
+  for(i in seq_along(fatFits)) {
+    if (!inherits(fatFits[[i]], c("simpleError", "error")) && is.null(funnelTestPlots[[.metamiscFitFunnelAsymmetryNames(fatFits[[i]]$method)]])) {
+
+      tempFunnelPlot   <- createJaspPlot(
+        title  = .metamiscFitFunnelAsymmetryNames(fatFits[[i]]$method),
+        width  = 340,
+        height = 300)
+      tempFunnelPlot$position <- i
+      tempFunnelPlot$dependOn(.metamiscFitFunnelAsymmetryOptions(fatFits[[i]]$method))
+      funnelTestPlots[[.metamiscFitFunnelAsymmetryNames(fatFits[[i]]$method)]] <- tempFunnelPlot
+
+      tempPlot <- tryCatch(.metamiscFitFunnelAsymmetryggPlot(fatFits[[i]]), error = function(e)e)
+
+      if (any(class(tempPlot) %in% c("simpleError", "error"))) {
+        tempFunnelPlot$setError(tempPlot$message)
+      } else{
+        tempPlot <- tempPlot + jaspGraphs::geom_rangeframe() + jaspGraphs::themeJaspRaw()
+        tempFunnelPlot$plotObject <- tempPlot
+      }
+
+    }
+  }
+
+  return()
+}
+.metamiscFitFunnelAsymmetryNames <- function(shortcut) {
+  switch(
+    shortcut,
+    "E-UW"  = "Egger (unweighted)",
+    "E-FIV" = "Egger (multiplicative overdispersion)",
+    "M-FIV" = "Macaskill",
+    "M-FPV" = "Macaskill (pooled)",
+    "P-FPV" = "Peters",
+    "D-FIV" = "Debray"
+  )
```

Comment:
Should these be translated? If yes, I would not use them for indexing in jaspResults.